airflow/providers/amazon/aws/transfers/sql_to_s3.py

```diff
@@ -194,13 +197,32 @@ def execute(self, context: Context) -> None:
 
     def _partition_dataframe(self, df: pd.DataFrame) -> Iterable[tuple[str, pd.DataFrame]]:
         """Partition dataframe using pandas groupby() method."""
+        try:
+            import secrets
+            import string
+
+            import numpy as np
+        except ImportError:
+            pass
+        # if max_rows_per_file argument is specified, a temporary column with a random unusual name will be
+        # added to the dataframe. This column is used to dispatch the dataframe into smaller ones using groupby()
+        random_column_name = ""
+        if self.max_rows_per_file and not self.groupby_kwargs:
```

Comment:
Could you please rather fail in case both parameters are specified? I think it's better to fail explicitly rather than silently swallow `max_ros_per_file`.

Reply:
Sure!